mifospay/src/main/java/org/mifos/mobilewallet/mifospay/qr/ui/ReadQrActivity.kt

```diff
@@ -0,0 +1,139 @@
+package org.mifos.mobilewallet.mifospay.qr.ui
+
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import com.google.zxing.*
+import com.google.zxing.common.HybridBinarizer
+import me.dm7.barcodescanner.zxing.ZXingScannerView
+import org.mifos.mobilewallet.mifospay.base.BaseActivity
+import org.mifos.mobilewallet.mifospay.databinding.ActivityReadQrBinding
+import org.mifos.mobilewallet.mifospay.qr.QrContract
+import org.mifos.mobilewallet.mifospay.qr.QrContract.ReadQrView
+import org.mifos.mobilewallet.mifospay.qr.presenter.ReadQrPresenter
+import org.mifos.mobilewallet.mifospay.utils.Constants
+import java.io.FileNotFoundException
+
+/**
+ * Created by naman on 7/9/17.
+ */
+class ReadQrActivity : BaseActivity(), ReadQrView, ZXingScannerView.ResultHandler {
+    private lateinit var mPresenter: ReadQrPresenter
```

Comment:
You did not use the Dagger field injection, this will lead to Null Pointer Exception later on.